session.go

```diff
@@ -825,6 +828,9 @@ type Credential struct {
 	// Mechanism defines the protocol for credential negotiation.
 	// Defaults to "MONGODB-CR".
 	Mechanism string
+
+	// Certificate defines an x509 certificate for authentication at login.
```

Comment:
The documentation here should specify the constraints - specifically: 

- The `Username` field is populated from the cert and should not be set
- The `Mechanism` field should be `MONGODB-X509` or not set (automatically populate in this case)
- The `Source` field should be `$external` or not set (auto populate too)